src/operation/select-rule-comparison.bench.ts

```
@@ -0,0 +1,302 @@
+    return { success: false };
+  }
+
+  // Find the best rule for full selection (original count <= limit)
+  const fullSelectableRules = ruleCounts.filter(
+    (rule) => rule.originalCount <= limitCount,
+  );
+
+  if (fullSelectableRules.length > 0) {
+    // Select rule with highest filtered count (old logic)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    let bestRule = fullSelectableRules[0]!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)